src/MultiSigInbox/MultiSigInboxPlugin.cs

```diff
@@ -0,0 +1,61 @@
+using Akka.Actor;
+using Neo.ConsoleService;
+using Neo.Wallets;
+
+namespace Neo.Plugins.MultiSigInbox
+{
+    public class MultiSigInboxPlugin : Plugin, IP2PPlugin
+    {
+        public const string StatePayloadCategory = "MultiSignatureInbox";
+        public override string Name => "MultiSigInbox";
+        public override string Description => "Enables MultiSigInbox for the node";
+
+        private IWalletProvider walletProvider;
+        private IActorRef service;
+        private bool started = false;
+
+        protected override void Configure()
+        {
+            Settings.Load(GetConfiguration());
+        }
+
+        protected override void OnPluginsLoaded()
+        {
+            walletProvider = GetService<IWalletProvider>();
+            if (Settings.Default.AutoStart)
+                walletProvider.WalletOpened += WalletProvider_WalletOpened;
+        }
+
+        private void WalletProvider_WalletOpened(object sender, Wallet wallet)
+        {
+            walletProvider.WalletOpened -= WalletProvider_WalletOpened;
+            Start(wallet);
+        }
+
+        [ConsoleCommand("start multisiginbox", Category = "MultiSigInbox", Description = "Start multiSigInbox service")]
+        private void OnStart()
+        {
+            Start(walletProvider.GetWallet());
+        }
+
+        public void Start(Wallet wallet)
+        {
+            if (started) return;
+            started = true;
+            service = System.ActorSystem.ActorOf(MultiSigInboxService.Props(System.LocalNode, System.Blockchain, System.LoadStore(Settings.Default.Path), wallet));
+            service.Tell(new MultiSigInboxService.Start());
+        }
+
+        [ConsoleCommand("inbox list", Category = "MultiSigInbox", Description = "List pending transactions")]
+        private void OnInboxList(bool pending = true)
```

Comment:
@shargon, I think that signers can send both a complete transaction that is partially signed or just the hash and a partial signature.
In this way we avoid unnecessary traffic.

Reply:
However, transactions are usually small and not really a problem for the P2P.
Perhaps all should send the complete tx.